test/common/common/re_speed_test.cc

```diff
@@ -0,0 +1,128 @@
+// Note: this should be run with --compilation_mode=opt, and would benefit from
+// a quiescent system with disabled cstate power management.
+
+#include <regex>
+
+#include "common/common/assert.h"
+
+#include "absl/strings/string_view.h"
+#include "benchmark/benchmark.h"
+#include "re2/re2.h"
+
+// NOLINT(namespace-envoy)
+
+static const char* cluster_inputs[] = {
+    "cluster.no_trailing_dot",
+    "cluster.match.",
+    "cluster.match.normal",
+    "cluster.match.and.a.whole.lot.of.things.coming.after.the.matches.really.too.much.stuff",
+};
+
+static const char cluster_re_pattern[] = "^cluster\\.((.*?)\\.)";
+static const char cluster_re_alt_pattern[] = "^cluster\\.(([^\\.]+)\\.).*";
```

Comment:
constants should follow the kNamingConvention.

Reply:
actually in Envoy it's just `NamingConvention`.